webapp/src/pages/Settings.tsx

```diff
@@ -207,6 +217,38 @@ const Settings: React.FC = () => {
     Partial<AzimuthConfig>
   >({});
 
+  const resultingConfig = Object.assign({}, config, partialConfig);
+
+  React.useEffect(() => {
+    if (defaultConfig && defaultConfig.language !== resultingConfig.language) {
+      setPartialConfig({
+        language: defaultConfig.language,
+        syntax: {
+          ...resultingConfig.syntax,
+          spacy_model: defaultConfig.syntax.spacy_model,
+          subj_tags: defaultConfig.syntax.subj_tags,
+          obj_tags: defaultConfig.syntax.obj_tags,
+        },
+        similarity: resultingConfig.similarity && {
+          ...(resultingConfig.similarity ?? defaultConfig.similarity!),
```

Comment:
Since you added `resultingConfig.similarity &&`, `resultingConfig.similarity` will always be defined, here.
```suggestion
          ...resultingConfig.similarity,
```